injection/sharedmain/main.go

```diff
@@ -482,3 +521,45 @@ func ControllersAndWebhooksFromCtors(ctx context.Context,
 
 	return controllers, webhooks
 }
+
+func getPodName() string {
+	// If PodName is injected into the env vars, set it on the logger.
+	// This is needed for HA components to distinguish logs from different
+	// pods.
+	if val := os.Getenv("POD_NAME"); val != "" {
+		return val
+	}
+
+	// CRI-O and containerd set HOSTNAME to the Pod Name.
+	// This seems to be K8s behaviour - originating from
+	// older docker behaviour
+	if val := os.Getenv("HOSTNAME"); val != "" {
+		return val
+	}
+	return ""
```

Comment:
Do we need the detailed comments here, or can we just do:

```suggestion
	return cmp.Or(os.Getenv("POD_NAME"), os.Getenv("HOSTNAME"), "")
```

Reply:
FYI POD_NAME is a knative convention using downward API that we use